File: lib/domain/cultivation_plan/ports/cultivation_plan_destroy_output_port.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Ports
      class CultivationPlanDestroyOutputPort
        def on_success(destroy_output_dto)
          raise NotImplementedError, "Subclasses must implement on_success"
        end

        def on_failure(error_dto)
          raise NotImplementedError, "Subclasses must implement on_failure"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/ports/cultivation_plan_destroy_output_port.rs
//! Ruby: `Domain::CultivationPlan::Ports::CultivationPlanDestroyOutputPort`

use crate::cultivation_plan::dtos::CultivationPlanDestroyOutput;
use crate::shared::dtos::Error;

pub trait CultivationPlanDestroyOutputPort {
    fn on_success(&mut self, dto: CultivationPlanDestroyOutput);
    fn on_failure(&mut self, error: Error);
}
